IOPool/Common/bin/EdmProvDump.cc

```diff
@@ -79,8 +79,18 @@ namespace {
     const_iterator end() const { return children_.end(); }
 
     void print(std::ostream& os) const {
-      os << config_.processName() << " '" << config_.passID() << "' '" << config_.releaseVersion() << "' [" << simpleId_
-         << "]  (" << config_.parameterSetID() << ")" << std::endl;
+      auto const& hwresources = config_.hardwareResourcesDescription();
+      os << config_.processName() << " '" << config_.releaseVersion() << "' [" << simpleId_ << "]  ("
+         << config_.parameterSetID() << ")"
+         << "  (" << hwresources.microarchitecture;
+      if (not hwresources.selectedAccelerators.empty()) {
+        os << "; " << hwresources.selectedAccelerators.front();
+        for (auto it = hwresources.selectedAccelerators.begin() + 1; it != hwresources.selectedAccelerators.end();
+             ++it) {
+          os << "," << *it;
+        }
+      }
+      os << ")" << std::endl;
```

Comment:
@makortel , crab uses `edmProvDump` to [find the hash](https://github.com/dmwm/CRABServer/blob/master/scripts/CMSRunAnalysis.py#L496) and it is sensitive to changes here e.g. crab tests for 15.1.X now [fail](https://cms-talk.web.cern.ch/t/crab-job-failed-pset-hash-missing-from-edmprovdump/116834/3). Should we add a comment here to make sure any future changes keep this in mind

Reply:
even better, avoid this fragility. What could be done instead to get the PSet hash ? CRAB uses it to enforce that users can't add to same dataset files produced with modified PSet.

Reply:
I agree with @belforte, we need a better (or, more stable) solution for CRAB. (e.g. I'm already working to change the `edmProvDump` even more, that would break the regex in question completely).

I need to first understand what exactly CRAB needs. Do I understand correctly that it is the hash of the PSet of the Process that produced the file?

Reply:
adding  CRAB DevOps  @aspiringmind-code and @sinonkt who will follow up on needed changes on CRAB side

Reply:
feel free to move more discussion out of this close PR. We already have a CRAB issue https://github.com/dmwm/CRABServer/issues/8953